src/metabase/api/database.clj

```diff
@@ -516,7 +516,7 @@
                 (m/update-existing details k (constantly (get-in database [:details k])))
                 details))
             details
-            database/sensitive-fields))))
+            (database/get-sensitive-fields-for-db database)))))
```

Comment:
I'd just do 

```clj
(driver/get-sensitive-fields (driver.u/database->driver database))
```

directly here, since the notion of "sensitive fields" isn't really specific to an individual database but rather to an individual driver... it doesn't make sense to write a utility function in `models.database` for what is effectively just the one-liner above

Reply:
Yeah this was mostly to dance around some weird tests that were passing in null and empty maps for DB specs.  I'll figure out another way to make them happy.